plugins/foreman_virt_who_configure/0.5/index.md

```diff
@@ -0,0 +1,272 @@
+
+```sh
+scp /home/user/deploy_virt_who_config_4.sh root@satellite.example.com:/root/
+ssh root@satellite.example.com
+chmod u+x /root/deploy_virt_who_config_4.sh
+/root/deploy_virt_who_config_4.sh
+```
+
+### 3.2.3 Using the API
+
+The plugin comes with a REST API that can be used. You can use the API endpoint for downloading the script directly on the host, where you want to deploy the virt-who. The following curl command would download it. Make sure you keep it in secure place.
+
+```sh
+curl -u admin:changeme  https://foreman.example.tst/foreman_virt_who_configure/api/v2//configs/4/deploy_script.sh > /root/deploy_script.sh
+chmod u+x /root/deploy_script.sh
+/root/deploy_script.sh
+```
+
+Obviously you need to change admin/changeme credentials according to your setup. The virt-who configuration id is also part of the url, 4 in this case.
+
+### 3.2.4 Using Hammer (recommended)
+
+[Hammer CLI](https://theforeman.org/manuals/latest/#3.5.6CLI) can be used to download and/or deploy the configuration script. Note that Hammer CLI can be installed on the target host and configured to run commands against your Foreman instance. We provide a plugin for Hammer CLI that must be installed in order for this to work. 
+
+All functionality that is provided in the WebUI can be achieved using this plugin. To see how to use the Hammer CLI plugin command, see
+
+```sh
+hammer virt-who-config -h
+```
+
+#### Listing configs
+
+```sh
+$ hammer virt-who-config list
+---|--------|---------------|--------|--------------------
+ID | NAME   | INTERVAL      | STATUS | LAST REPORT AT     
+---|--------|---------------|--------|--------------------
+2  | RHV    | every 4 hours | OK     | 2021/10/12 20:13:54
+1  | VMware | every 4 hours | OK     | 2021/10/14 16:14:09
+---|--------|---------------|--------|--------------------
+```
+
+#### Deploying a configuration
+ 
+```sh
+hammer virt-who-config deploy --id 4
+```
+
+This will download the script for configuration with id 4 to a secure location, run the script and it prints the result to STDOUT. You can also use name of configuration to identify which virt-who configuration you want to deploy:
+ 
+```sh
+hammer virt-who-config deploy --name example_config
+```
+
+If you wish to review the script first, you can download it by the following Hammer CLI command:
+
+```sh
+hammer virt-who-config fetch --id 4
+hammer virt-who-config fetch --id 4 --output /root/script.sh
+```
+
+The first example prints the configuration script on STDOUT, second stores it into the specified path. Hammer CLI also checks that the file does not exist previously to avoid a race condition security risk. It also configures permissions properly so only the current user can read and execute it. In both cases, `--name` could be used instead of `--id` 
```

Comment:
```suggestion
The first example prints the configuration script on STDOUT; the second stores it into the specified path. Hammer CLI also checks that the file does not exist previously to avoid a race condition security risk. It also configures permissions properly so only the current user can read and execute it. In both cases, `--name` could be used instead of `--id`. 
```